ixwebsocket/IXWebSocketTransport.cpp

```diff
@@ -565,6 +573,15 @@ namespace ix
             _rxbuf.erase(_rxbuf.begin(),
                          _rxbuf.begin() + ws.header_size + (size_t) ws.N);
         }
+
+        // if an abnormal closure was raised, and nothing else triggered a CLOSED state in
+        // the received and processed data, then close using abnormal close code and message
+        if (_readyState == CLOSING && _treatAbnormalCloseAfterDispatch)
+        {
+            _treatAbnormalCloseAfterDispatch = false;
+            
```

Comment:
Why are we sure that remote is true here ? Because we only get there if reading from the socket returned a negative value in poll ?

Reply:
Yes